File: 349-intersection-of-two-arrays/intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        # lst = [x for x in nums1 if x in nums2]
        # lst_set = set(lst)
        # result = list(lst_set)
        # return result
        return list( set(nums1) & set(nums2) )   # & --> Intersection

File: 349-intersection-of-two-arrays/intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> set;
        for(auto it : nums1){
            set.insert(it);
        }
        vector<int> ans;

        for(auto it : nums2){
            if(set.erase(it)) ans.push_back(it);
        }
        return ans;
    }
};